### ISSUE TITLE
Allow enable/disable all via config file

### ISSUE THREAD

**Issue by anderseknert** [StyraInc/regal#153]
Since some users may want to only run a few rules in their repo, having to manually disable all the rules they don't care for is a bit of a chore. It'd be good if we allowed enabling/disabling all, or per category from the config as well.

**Comment by anderseknert** [comment]
Expanding on the scope a bit here as this was requested today, or at least something that is close enough. It'd be good if there was a way to freeze configuration of rules at any given point, so that updates wouldn't necessarily mean new rules automatically enabled. Some prefer to have them enabled manually, and that's obviously fine. I imagine this would look like:

1. Add a command to print the de-facto configuration used, i.e. the built-in default configuration merged with the user's custom config file.
2. Add a mode to Regal (settable via cia CLI args or config file) that have rules not mentioned in config ignored.
3. Log any rules ignored above so that it's easy for a user to see which rules they are missing from having frozen their rules config.

Definitely open to alternative approaches if anyone has suggestions!

**Comment by anderseknert** [comment]
I've been thinking some more about this, and I think a better design would be to mirror the `regal lint` flags for disabling/enabling all rules, or all per category, and allow more specific overrides, but via configuration rather than CLI args. I imagine it might look something like this:

```yaml
rules:
  '*':
    level: ignore
  bugs:
    '*':
      level: error
    constant-condition:
      level: ignore
  testing:
    print-or-trace-call:
      level: error
```

Disabling (i.e. setting level to `ignore` for) all rules at the top level, and then allow more specific overrides to enable (setting level to `warning` or `error` for) categories or rules would provide a way to "freeze" configuration for those who don't want new Regal versions to automatically add new rules, and it would also be a pretty convenient way to easily enable/disable all, categories and individual rules.

I'm not too fond of using `'*'` to mean "all" as the required quoting makes it rather ugly... but I'm not sure what would be a better option. We could call it "all" or "default", but it could easily be mistaken for a category/rule name. `'*'` while ugly at least clearly stands out from regular names. So might be that I go with that, but if anyone has better ideas, I'm listening 🙂

**Comment by anderseknert** [comment]
Fixed in #587
